google/auth/credentials.py

```diff
@@ -422,3 +480,16 @@ def signer(self):
         # pylint: disable=missing-raises-doc
         # (pylint doesn't recognize that this is abstract)
         raise NotImplementedError("Signer must be implemented.")
+
+
+class TokenState(Enum):
+    """
+    Tracks the state of a token.
+    FRESH: The token is valid. It is not expired or close to expired, or the token has no expiry. To make it mutually exclusive to STALE.
```

Comment:
sorry about this: the "To make it mutually exclusive to STALE" part wasn't meant to be part of the comment.